### PR TITLE
add endpoint to list all posts of a user

### PR DESCRIPTION
**Pull request by altontonn**
# Blog app: add API endpoints 💎 

In this PR, we worked on the following:-

- Created an API endpoint to list all posts for a user.

- Created an API endpoint to list all comments for a user's post.

- Created an API endpoint to add a comment to a post. Remember that the comment owner is the user that makes it; it should not be set by passing a parameter.

- Had API endpoints receive JSON and respond to JSON as well.

### Instructions for API Endpoint usage

1. In the URL add `http://localhost:3000/api/v1/users/16/posts/5/comments`
2. In the URL add `http://localhost:3000/api/v1/users/16/posts`
3. In the URL add `http://localhost:3000/api/v1/users`

### CODE REVIEW

**Review by Whoistolu — COMMENTED** [review]
Hi Team,
@UsirusenIkon, @altontonn and @RichardChileya 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a pair-programming project. Kindly ensure that hat the third developer on this project contributes equally. This is to encourage and groom you on how to work in a team. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class UsersController < ApplicationController
   def index
     @users = User.all
+    # render json: @users
   end
 
   def show
```
Comment (MaganaAsiati) [inline app/controllers/users_controller.rb:7]: - Kindly make sure that all API endpoints receive JSON and respond in a JSON format . as I see here your comment on the code that must return the JSON format please fix that :+1:  and make sure that you create the endpoint to return the list of users and is working as expected . to do that I can suggest you create a separate API file in your control for all and point :+1:

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -29,6 +32,12 @@ def show
     @post = Post.find(params[:id])
     @posts = @user.posts.find(params[:id])
     @comments = Comment.where(post_id: params[:id])
+    respond_to do |format|
+      format.html
+      format.xml { render xml: @post }
+      format.json { render json: @post }
+      format.json { render json: @comments }
+    end
```
Comment (MaganaAsiati) [inline app/controllers/posts_controller.rb:34]: - You did a great job by converting the response into JSON format but make sure that you have  API endpoint that returns all posts in JSON format- **GET:** `/api/users/:user_id/posts` or `/api/v1/users/:user_id/posts` as required in the project . try to test your points before submission so you will be sure there are working as expected. you can use thunder client or postman to test them :+1:

File: config/routes.rb [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
   resources :users, only: [:index, :show] do
     resources :posts, only:[:index, :show, :new, :create, :destroy] do
-      resources :comments, only:[:new, :create, :destroy]
+      resources :comments, only:[:index, :new, :create, :destroy]
       resources :likes, only:[:create]
     end
   end
```
Comment (MaganaAsiati) [inline config/routes.rb:12]: - Kindly make sure you have
   -  API endpoint that returns all comments for a post - **GET :** `/users/:user_id/posts/:id/comments`
   -  API endpoint that adds' a comment to a post - **POST :** `/users/:user_id/posts/:id/comments`
and your endpoints are working as expected. Please the your them before submission and give the documentation in the pull request on how your endpoints are working :+1:

**Review by MaganaAsiati — CHANGES_REQUESTED** [review]
# STATUS: Required Changes ♻️♻️♻️ 
## Hi @altontonn @UsirusenIkon  @RichardChileya  , I'm @MaganaAsiati, your reviewer for this PR  :wave: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close-gif-4

## Highlights

- Descriptive README file  :heavy_check_mark:👨🏻‍💻
- Great work using the right GitHub Flow.:heavy_check_mark:
- No Linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Kindly make sure you create the API and your endpoint is working as expected before submission make sure that you test your pin and you give the small documentation on how to use them in your pull request so  that it will be easy for the reviewer to test them :+1: 
- [ ] Kindly make also sure you update the readme file with your information :+1:  It will make your project looks more professional :+1: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._
- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future. :heavy_check_mark:👨🏻‍💻
-[ ] [OPTIONAL] kindly  Add a meaningful description to your pull request in the future which gives more information about what you implement in this milestone :heavy_check_mark:👨🏻‍💻


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by MaganaAsiati — APPROVED** [review]
# Status: Approved [✔️](https://emojipedia.org/check-mark/)
## Hi @altontonn @Whoistolu @UsirusenIkon   , I'm @MaganaAsiati , your reviewer for this PR  :wave: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


clap

## Highlights
- descriptive README file  :heavy_check_mark:👨🏻‍💻
- Your commits should start with a capital letter :heavy_check_mark:👨🏻‍💻


## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] [OPTIONAL] Please make sure to make your commit messages more descriptive in the future.
- [ ] [OPTIONAL] kindly  Add a meaningful description in your pull request in future which give more information about what you implement in this milestone :heavy_check_mark:👨🏻‍💻


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Cheers and Happy coding!👏👏👏

**Pull request merged**